getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,160 @@
+    Emoji.ANGER to "anger.png",
+    Emoji.SURPRISE to "surprise.png",
+    Emoji.SORROW to "sorrow.png",
+    Emoji.NONE to "none.png"
+)
+
+//Returns best emoji based on detected emotions likelihoods
+fun bestEmoji(annotation: FaceAnnotation): Emoji {
+    val emotionsLikelihood: Array<Likelihood> = arrayOf(Likelihood.VERY_LIKELY, Likelihood.LIKELY, Likelihood.POSSIBLE)
+    val emotions = mapOf(
+        Emoji.JOY to annotation.joyLikelihood,
+        Emoji.ANGER to annotation.angerLikelihood,
+        Emoji.SURPRISE to annotation.surpriseLikelihood,
+        Emoji.SORROW to annotation.sorrowLikelihood
+    )
+
+    for (emotionLikelihood in emotionsLikelihood) {
+        for (emotion in emotions) {
+            if (emotion.value == emotionLikelihood) return emotion.key
+        }
+    }
+    return Emoji.NONE
+}
+
+//Downloads source image to a temp file
+fun downloadObject(objectName: String) {
+    val blob = bucket.get(objectName)
+    val localFilePath = Paths.get("/tmp/$objectName")
+    val writeTo = PrintStream(FileOutputStream(localFilePath.toFile()))
+    writeTo.write(blob.getContent())
+    writeTo.close()
+}
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): String {
+
+        var publicUrl:String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+
+        val requests = ArrayList<AnnotateImageRequest>()
+        //Setting up image annotation request
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("$bucketUri/$objectName").build() //Fetching our source image
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        val responses = response.responsesList
+        var output = "" //kind of log
+
+        for (resp in responses) {
+            if (resp.hasError()) {
+                output += "Error: ${resp.error.message} \n"
+                return output
+            }
+
+            downloadObject(objectName) //image is now downloaded to /tmp/<objectName>
+            val imgBuff = ImageIO.read(Paths.get("/tmp/$objectName").toFile())
+            val gfx = imgBuff.createGraphics()
+            val extension = (Paths.get("/tmp/$objectName").toFile().extension)
+
+            for (annotation in resp.faceAnnotationsList) {
+                val emoji = bestEmoji(annotation)
+                val imgEmoji = ImageIO.read(Paths.get(EMOJIS_PATH + emojiPic[emoji]).toFile())
+
+                output += """
```

Comment:
Consider using a logger: https://cloud.google.com/appengine/docs/standard/java/logs/

Not only is it more efficient, but allows you to classify streams as DEBUG/INFO/WARNING/CRITICAL so that you can  view the level you need to.